### ISSUE TITLE
Some navigations not discovered in a cyclical model

### ISSUE THREAD

**Issue by AndriySvyryd** [dotnet/efcore#9205]
```C#
public class PostContext : DbContext
{
    public PostContext(DbContextOptions options)
        : base(options)
    {
    }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.Entity<MyPost>();
    }
}

public class MyBlog
{
    public int Id { get; set; }
    public List<MyPost> Posts { get; set; }

    public int OwnerId { get; set; }
    public Person Owner { get; set; }
}

public class MyPost
{
    public int Id { get; set; }

    public int BlogId { get; set; }
    public MyBlog Blog { get; set; }

    public int AuthorId { get; set; }
    public Person Author { get; set; }
}

public class Person
{
    public int Id { get; set; }
    public string Name { get; set; }
    public List<MyPost> AuthoredPosts { get; set; }
    public List<MyPost> OwnedBlogs { get; set; }
}
```

**Comment by ajcvickers** [comment]
@AndriySvyryd Thanks for investigating. We need to figure out ASAP if we want to take this in 2.0. Do you know what are the necessary conditions are to hit this--is it going to be common? Also, what are the workaround?
/cc @divega @Eilon Heads up that we may need to take a fix for this in 2.0.

**Comment by Eilon** [comment]
Ok, let me + @DamianEdwards know if we end up wanting to take this.
